clients/js/src/types.ts

```diff
@@ -145,15 +173,40 @@ export type ModifyCollectionParams = {
   metadata?: CollectionMetadata;
 };
 
-export type QueryParams = {
-  queryEmbeddings?: Embedding | Embeddings;
+/** This type represents the different ways the user can express a query for documents
+ *  - string: a simple text query which will be converted to an Embedding
+ *  - Embedding: a list of numbers representing the embedding of the query
+ *  - QueryDoc: a document with an embedding or contents field
+ */
+export type DocQuery = string | Embedding | QueryDoc;
+
+export type SingleQueryParams = {
   nResults?: PositiveInteger;
   where?: Where;
-  queryTexts?: string | string[];
+  query: DocQuery;
```

Comment:
not necessarily opposed to this, but wondering how much we want to try to keep Python/JS interfaces roughly in sync

@atroyn wdyt

Reply:
I think for this PR, we want to keep them in sync. This might be a bit janky for now but I would prefer that we review the interface as a whole between both clients rather than changing it piecemeal in each.

Can we create a new issue to discuss this?

Reply:
https://github.com/chroma-core/chroma/issues/2483

Reply:
@atroyn does that apply to the pluralization logic as well? I've added convenience overrides for singular document operations e.g:

```
const resp = getRecords(collection, {
  id: "blah"
});

console.log(resp); // prints "{id: "blah", embedding: [1, 2, 3], document: "whatever" }"
```